File: app/src/main/java/com/better/alarm/logger/SysoutLogWriter.java
package com.better.alarm.logger;

import com.better.alarm.logger.Logger.LogLevel;

public class SysoutLogWriter implements Logger.LogWriter {
    @Override
    public void write(LogLevel level, String tag, String message, Throwable e) {
        switch (level) {
            case INF:
            case DBG:
                System.out.println(tag + ", " + message);
                if (e != null) e.printStackTrace();
                break;

            case WRN:
            case ERR:
            default:
                System.err.println(tag + ", " + message);
                if (e != null) e.printStackTrace();
                break;
        }
    }
}

File: app/src/main/java/com/better/alarm/logger/SysoutLogWriter.kt
package com.better.alarm.logger

import com.better.alarm.logger.Logger.LogLevel

class SysoutLogWriter : LogWriter {
    override fun write(level: LogLevel, tag: String, message: String, e: Throwable?) {
        when (level) {
            LogLevel.INF, LogLevel.DBG -> {
                println("$tag, $message")
                e?.printStackTrace()
            }

            LogLevel.WRN, LogLevel.ERR -> {
                System.err.println("$tag, $message")
                e?.printStackTrace()
            }
        }
    }
}
